### PR TITLE
[common-1]: Create generic ApiResponse wrapper

### PR DESCRIPTION
**Pull request by Reyzis2021**
1. Task's name:
[common-1] Сreate generic ApiResponse wrapper

2. Developer's full name:
Konstantin Nyamtsu

3. Link to the PR: 
https://github.com/Sunagatov/Online-Store/pull/62

4. Link to Documentation:
https://github.com/Sunagatov/Online-Store/wiki/Api-Specification-for-Payment

5. Changes:
- Added generic ApiResponse wrapper

6. Tests:
The change was tested with using postman in the form of a universal json response from endpoints.

7. Test data:
Use any endpoint in the project and check response body of the endpoint

8. Conclusion:
Project was built with Apache Maven successfully.
Project was tested manually with Postman successfully.

### CODE REVIEW

**Review by UladzislauM — APPROVED** [review]

File: src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+public record ApiResponse<T>(
+
+        @JsonProperty("data")
+        T data,
+
+        @JsonProperty("message")
+        String message,
+
+        @JsonProperty("status")
+        Integer status,
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java:17]: httpStatusCode

File: src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+public record ApiResponse<T>(
+
+        @JsonProperty("data")
+        T data,
+
+        @JsonProperty("message")
+        String message,
+
+        @JsonProperty("status")
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java:16]: httpStatusCode

File: src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        Integer status,
+
+        @JsonProperty("time_stamp")
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java:19]: it must be camelCase.
 `@JsonProperty("timestamp")`

File: src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
+        LocalDateTime timeStamp
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java:21]: timestamp is the one word
`LocalDateTime timestamp`

File: src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        T data,
+
+        @JsonProperty("message")
+        String message,
+
+        @JsonProperty("status")
+        Integer status,
+
+        @JsonProperty("time_stamp")
+        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java:20]: Please, create TIMESTAMP_JSON_FORMAT constant in `com.zufar.onlinestore.common.response;` and use it in this class.
```
public static final String TIMESTAMP_JSON_FORMAT = "yyyy-MM-dd HH:mm:ss";
```

File: src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java (hunk 3 above)
Comment (Reyzis2021) [reply src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java:19]: Done

File: src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java (hunk 2 above)
Comment (Reyzis2021) [reply src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java:16]: Done

File: src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java (hunk 1 above)
Comment (Reyzis2021) [reply src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java:17]: Done

File: src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java (hunk 4 above)
Comment (Reyzis2021) [reply src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java:21]: Done

File: src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java (hunk 5 above)
Comment (Reyzis2021) [reply src/main/java/com/zufar/onlinestore/common/response/ApiResponse.java:20]: Done

**Review by Sunagatov — APPROVED** [review]

**Review by vitaliibredun — APPROVED** [review]
Approved

**Review by yevr19 — APPROVED** [review]

**Review by Sunagatov — APPROVED** [review]

**Pull request merged**